### ISSUE TITLE
Can't construct typed DataTable based on SqlCommandProvider

### ISSUE THREAD

**Issue by smoothdeveloper** [fsprojects/FSharp.Data.SqlClient#228]
``` fsharp
type MyCommand = SqlCommandProvider<..., ResultType = DataTable>

let table = MyCommand.Table() // doesn't compile
```

I think this should be available, and eventually the table should have all enabled for manual population.

I'm using this because I'm calling a stored procedure (with same output schema) using dynamic sql and UDDT, the stored procedure output has same schema as a specific command, but just to get an instance of such table, I need to go through ugly hoops.

**Comment by dmitry-a-morozov** [comment]
How about statically typed ```Clone``` method?

**Comment by smoothdeveloper** [comment]
@dmitry-a-morozov to call a `Clone` you need a first instance available right?

I'm actually trying to get that first empty instance and to do so, right now I need to run a SQL query which returns 0 records with expected schema.

If we expose public constructor, we could as well expose a statically typed `Clone`, I'd be happy to investigate this further if you think the feature is useful.

**Comment by dmitry-a-morozov** [comment]
I really think that ctor on `SqlCommandProvider` generated data tables is flawed design. But `Clone` is there anyway. 
To get empty result set and not screw up command text you can consider to call `set fmtonly on` before first execution. Then set back to `set fmtonly off`

**Comment by smoothdeveloper** [comment]
Thanks for the hint regarding `set fmtonly`, it seems to me the most straightforward would still be to expose a constructor, this avoids any roundtrip with database and just reuse the information determined at compile time, and it is a single line of code to get the object ready.

**Comment by smoothdeveloper** [comment]
I've made some progress and have the code to make this work, if there are people interested in this feature, please show up in the thread.

**Comment by smoothdeveloper** [comment]
Since I've made progress on that and the feature start to be necessary in my case (I can't write unit tests against typed datatables without hitting the database right now), I'll prepare a PR for it.

**Comment by smoothdeveloper** [comment]
fixed via #303
